reana_job_controller/job_monitor.py

```diff
@@ -469,6 +480,138 @@ def watch_jobs(self, job_db, app=None):
                 time.sleep(120)
 
 
+@singleton
+class JobMonitorCompute4PUNCH(JobMonitor):
+    """HTCondor jobs monitor Compute4PUNCH."""
+
+    def __init__(self, **kwargs):
+        """Initialize Compute4PUNCH job monitor thread."""
+        self.job_manager_cls = COMPUTE_BACKENDS["compute4punch"]()
+        super(__class__, self).__init__(thread_name="compute4punch_job_monitor")
+
+    def watch_jobs(self, job_db, app=None):
+        """
+        Use SSH connection to Compute4PUNCH login node to monitor jobs.
+
+        :param job_db: Dictionary which contains all running jobs.
+        """
+        c4p_connection = SSHClient(
+            hostname=C4P_LOGIN_NODE_HOSTNAME,
+            port=C4P_LOGIN_NODE_PORT,
+            timeout=C4P_SSH_TIMEOUT,
+            banner_timeout=C4P_SSH_BANNER_TIMEOUT,
+            auth_timeout=C4P_SSH_AUTH_TIMEOUT,
+            auth_strategy=motley_cue_auth_strategy_factory(
+                hostname=C4P_LOGIN_NODE_HOSTNAME
+            ),
+        )
+
+        while True:
+            logging.debug(
+                "Starting a new stream request to watch Jobs on Compute4PUNCH"
+            )
+            try:
+                c4p_job_mapping = {
+                    job_dict["backend_job_id"]: reana_job_id
+                    for reana_job_id, job_dict in job_db.items()
+                    if filter_jobs_to_watch(
+                        reana_job_id, job_db, compute_backend="compute4punch"
+                    )
+                }
+                c4p_job_statuses = query_c4p_jobs(
+                    *c4p_job_mapping.keys(), ssh_client=c4p_connection
+                )
+                logging.debug(f"Compute4PUNCH JobStatuses: {c4p_job_statuses}")
```

Comment:
What do you think about using `logging.info` instead of `logging.debug` here in the job monitor? By default debug-level logs are not shown by REANA, but these are really useful when a running job gets "stuck" to understand what went wrong.

Reply:
Thanks, fixed in https://github.com/reanahub/reana-job-controller/pull/430/commits/5024987999f67e216d422b61d1cb25ac71974e58